WordPressKit/WordPressKit/WordPressOrgXMLRPCApi.swift

```diff
@@ -189,10 +189,9 @@ open class WordPressOrgXMLRPCApi: NSObject {
                         DispatchQueue.main.async {
                             success(responseObject, response.response)
                         }
-                    } catch let error as NSError {
-                        progress.completedUnitCount = 0
```

Comment:
Is there a good reason for removing resetting the completed unit count to zero? I originally added this to fix some issues surrounding reporting errors – the original issue is [here](https://github.com/wordpress-mobile/WordPress-iOS/pull/8407), but a summary of the main problem is:

> * First, within the Rest API we were updating the upload's progress, then calling the failure or completion blocks.
> * When the upload had either completed or failed, the progress would be set to 100%.
> * This would trigger the progress observer in MediaProgressCoordinator. Because the failure block hadn't yet been called, we wouldn't have attached an error to the failed uploads, but the progress coordinator would detect the upload as complete / successful because the progress was 100%.
> 
> I changed this behaviour so that in the case of a failure we set the progress to 0%. Then, when the failure block is called, we'll attach the error and check if all of the uploads have completed or failed. If so, the progress coordinator is not running any more.

Note that if we removed this setting progress to 0 here, the REST API still does it on failure so we'd have a mismatch in the behaviour of XMLRPC and REST.

Reply:
I think we should keep whatever value we have at the moment that we had the failure.
This is related to the change above where progress is still being reported after the failure, this is another trigger for a race condition between the reporting of progress and reporting of failure.